### PR TITLE
engine: common: minor code fixes in hpak.c

### PR DESCRIPTION
**Pull request by SNMetamorph**
Including fix of bug that causes writing crap in custom.hpk
изображение

### CODE REVIEW

File: engine/common/hpak.c [hunk 1]
```diff
@@ -512,7 +512,7 @@ void HPAK_CheckSize( const char *filename )
 	Q_strncpy( pakname, filename, sizeof( pakname ));
 	COM_ReplaceExtension( pakname, ".hpk" );
 
-	if( FS_FileSize( pakname, false ) > ( maxsize * 1000000 ))
+	if( FS_FileSize( pakname, false ) > ( maxsize * 1048576 ))
```
Comment (a1batross) [inline engine/common/hpak.c:515]: What's this magical number? 1024 * 1024?

File: engine/common/hpak.c [hunk 2]
```diff
@@ -17,8 +17,8 @@ GNU General Public License for more details.
 #include "hpak.h"
 
 #define HPAK_MAX_ENTRIES	0x8000
-#define HPAK_MIN_SIZE	(1 * 1024)
-#define HPAK_MAX_SIZE	(128 * 1024)
+#define HPAK_ENTRY_MIN_SIZE	(512)
```
Comment (a1batross) [inline engine/common/hpak.c:20]: Do we want to have minimal entry size here?

File: engine/common/hpak.c [hunk 3]
```diff
@@ -325,7 +325,7 @@ void HPAK_AddLump( qboolean bUseQueue, const char *name, resource_t *pResource,
 	// make a new container
 	dstpak.count = srcpak.count + 1;
 	dstpak.entries = Z_Malloc( sizeof( hpak_lump_t ) * dstpak.count );
-	memcpy( dstpak.entries, srcpak.entries, srcpak.count );
+	memcpy( dstpak.entries, srcpak.entries, sizeof( hpak_lump_t ) * srcpak.count );
```
Comment (a1batross) [inline engine/common/hpak.c:328]: Nice catch!

File: engine/common/hpak.c (hunk 2 above)
Comment (SNMetamorph) [reply engine/common/hpak.c:20]: You mean we just can set minimal entry size 0 and just remove this macro?

File: engine/common/hpak.c (hunk 1 above)
Comment (SNMetamorph) [reply engine/common/hpak.c:515]: Yes, it's 1024 * 1024, 1 megabyte

**Comment by a1batross** [comment]
Nah, actually, it's fine

**Review by a1batross — APPROVED** [review]

**Pull request merged**